src/overlaybd/zfile/compressor.cpp

```diff
@@ -192,69 +197,173 @@ class LZ4Compressor : public BaseCompressor {
         return (qat_enable ? DEFAULT_N_BATCH : 1);
     }
 
-    virtual int do_compress(size_t *src_chunk_len, size_t *dst_chunk_len,
-                            size_t dst_buffer_capacity, size_t nblock) override {
+    int do_compress(size_t *src_chunk_len, size_t *dst_chunk_len, size_t dst_buffer_capacity,
```

Comment:
Usually we don't need to put the function body backward.

Reply:
I don't feel necessary to have separate do_compress() and do_decompress(). Simply realizes the logic in batch_compress() / batch_decompress(). I do feel necessary to have qat-related logic separated as a singleton object shared by all lz4_decompressor objects (and threads). The qat wrapper object simply provides 2 functions: batch_compress() / batch_decompress(). Whenever they fail, we fall back to the software-only implementation.